lib/peatio/injectors/peatio_events.rb

```diff
@@ -60,39 +66,98 @@ def public_orderbook
         "update",
         {
           "asks": [
-            ["1020.0","0.005"],
-            ["1026.0","0.03"]
+            ["1020.0", "0.005"],
+            ["1026.0", "0.03"]
           ],
           "bids": [
-            ["1000.0","0.25"],
-            ["999.0","0.005"],
-            ["994.0","0.005"],
-            ["1.0","11.0"]
+            ["1000.0", "0.25"],
+            ["999.0", "0.005"],
+            ["994.0", "0.005"],
+            ["1.0", "11.0"]
           ]
         }
       ]
     end
 
+    def public_orderbook_snapshot1
+      [
```

Comment:
subscribe to channel
receive the book snapshot and create your in-memory book structure
when count > 0 then you have to add or update the price level
3.1 if amount > 0 then add/update bids
3.2 if amount < 0 then add/update asks
when count = 0 then you have to delete the price level.
4.1 if amount = 1 then remove from bids
4.2 if amount = -1 then remove from asks

```
[
  CHANNEL_ID,
  [
    [
      PRICE,
      COUNT,
      AMOUNT
    ],
    ...
  ]
]
  
// on funding currencies (ex. fUSD)
[
  CHANNEL_ID,
  [
    [
      RATE,
      PERIOD,
      COUNT,
      AMOUNT
    ],
    ...
  ]
]
```